gh-ci-pipeline/src/main/java/Main.java

```diff
@@ -0,0 +1,190 @@
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.security.NoSuchAlgorithmException;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+import com.amazonaws.services.kms.model.DecryptRequest;
+import com.amazonaws.services.kms.model.EncryptRequest;
+import com.amazonaws.services.kms.model.ReEncryptRequest;
+
+public class Main {
+
+    // From slides -------------------------------------
+    static String myKeyId = "AKIAX742FUDUQXXXXXXX";
+    static String mySecretKey = "MY_SECRET_KEY";
+    static final String CIPHER = "DES";
+    static final String CIPHER_GOOD ="RSA/ECB/OAEPPadding";
+
+    static final String VALID_PATH1 = "./test/file1.txt";
+    static final String VALID_PATH2 = "./test/file2.txt";
+    static final String DEFAULT_VALID_PATH = "./test/file3.txt";
+
+    public static void main(String[] args) {
+        AWSCredentials creds =
+            getCreds(myKeyId, mySecretKey);
+        System.out.println(creds.getAWSSecretKey());
+
+        AWSCredentials creds2 = getCreds();
+        System.out.println(creds2.getAWSSecretKey());
+
+        run1();
+        run2();
+
+        reEncrypt1();
+        reEncrypt2();
+    }
+
+    // Bad
+    static AWSCredentials getCreds(String id, String key) {
+        return new BasicAWSCredentials(id, key);
+    }
+
+    // Good
+    static AWSCredentials getCreds() {
+        DefaultAWSCredentialsProviderChain creds =
+            new DefaultAWSCredentialsProviderChain();
+        return creds.getCredentials();
+    }
+
+    // Bad
+    public static void run1() {
+        try {
+            Cipher cipher = Cipher.getInstance(CIPHER);
+            System.out.println(cipher);
+        }
+        catch (NoSuchPaddingException e) {
+            e.printStackTrace();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 62:70, 75:82.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.